<launch>
  <!-- R2, SE2, R3, or SE3 are supported -->
  <arg name="sampling_space" default="R2" />

  <node pkg="differentiable_rmap" type="NodeRmapPlanning" name="rmap_planning"
        output="screen">
    <rosparam subst_value="true">
      sampling_space: $(arg sampling_space)
      config_path: $(find differentiable_rmap)/config/RmapPlanning.yaml
      svm_path: /tmp/rmap_svm_model_$(arg sampling_space).libsvm
    </rosparam>
  </node>

  <node pkg="optmotiongen" type="InteractiveMarkerManager.py" name="rviz_server"
        clear_params="true">
    <rosparam if="$(eval arg('sampling_space') == 'R2')">
      interactive_markers:
        - name: target
          initial_pose:
            pos: [0.0, 0.0, 99.0]
            rot: [1.0, 0.0, 0.0, 0.0]
          controls: [tx, ty]
          markers:
            - type: 2 # SPHERE
              scale: [0.15, 0.15, 0.15]
              color: [0.0, 1.0, 0.0, 1.0]
    </rosparam>
    <rosparam if="$(eval arg('sampling_space') == 'SE2')">
      interactive_markers:
        - name: target
          initial_pose:
            pos: [0.0, 0.0, 99.0]
            rot: [1.0, 0.0, 0.0, 0.0]
          controls: [tx, ty, rz]
          markers:
            - type: 2 # SPHERE
              scale: [0.15, 0.15, 0.15]
              color: [0.0, 1.0, 0.0, 1.0]
            - type: 0 # ARROW
              scale: [0.3, 0.1, 0.1]
              color: [0.0, 1.0, 0.0, 1.0]
    </rosparam>
  </node>

  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find differentiable_rmap)/rviz/RmapPlanning.rviz"
        output="log" clear_params="true" />
</launch>
